### DISCUSSION TITLE
Storage upload unauthorized using xc-token authentication header(it works with xc-auth)

### DISCUSSION THREAD

**Discussion by rst10124492** [nocodb/nocodb#5712] [Q&A] [answered]
Did the storage API changed in the last version I'm getting `{'msg': 'Unauthorized'}` when trying to upload a file to storage using `xc-token`, the upload works fine when using `xc-auth`

This worked in the last version, not sure if I should report it as a bug or I'm missing something.

And yes, the xc-token is valid.

**Comment by wingkwong** [comment]
Auth Token (`xc-auth`) is a JWT Token generated based on the logged-in user. By default, the token is only valid for 10 hours. However, you can change the value by defining it using environment variable `NC_JWT_EXPIRES_IN`. I think you are using an expired `xc-auth` which caused unauthorized access. 

On the other hand, `xc-token` doesn't have expiry date. Therefore, if you have third-party applications using our APIs, then it's recommended to use `xc-token` instead `xc-auth`. If you insist to use `xc-auth`, then you need to get the new auth token after it expires.

**Reply by rst10124492** [reply]
Thank you for answering but you did not read my question properly.

>then it's recommended to use xc-token instead xc-auth

But I am using xc-token and IT DOES NOT WORK for storage API while working for everything else.

That is my issue, with a valid xc-token the storage api does not work.

**Reply by wingkwong** [reply]
Oh sorry - I misread it. Found the root cause. Will fix it asap and include in the today release.

**Reply by wingkwong** [reply, marked as answer]
Fixed [here](https://github.com/nocodb/nocodb/pull/5720) - will be available in `0.107.4`.
